transformer/dockerfilegenerator/java/waranalyser.go

```diff
@@ -70,24 +70,29 @@ func (t *WarAnalyser) GetConfig() (transformertypes.Transformer, *environment.En
 // DirectoryDetect runs detect in each sub directory
 func (t *WarAnalyser) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
 	services = map[string][]transformertypes.Artifact{}
-	warFilePaths, err := common.GetFilesInCurrentDirectory(dir, nil, []string{".*[.]war"})
+	paths, err := common.GetFilesInCurrentDirectory(dir, nil, []string{".*[.]war"})
 	if err != nil {
 		logrus.Errorf("Error while parsing directory %s for jar file : %s", dir, err)
 		return nil, err
 	}
-	if len(warFilePaths) == 0 {
+	if len(paths) == 0 {
 		return nil, nil
 	}
-	for _, path := range warFilePaths {
+	for _, path := range paths {
+		relPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), path)
+		if err != nil {
+			logrus.Errorf("failed to make the path %s relative to the sourc code directory %s . Error: %q", path, t.Env.GetEnvironmentSource(), err)
+			continue
+		}
 		newArtifact := transformertypes.Artifact{
 			Paths: map[transformertypes.PathType][]string{
 				artifacts.WarPathType:        {path},
 				artifacts.ServiceDirPathType: {filepath.Dir(path)},
 			},
 			Configs: map[transformertypes.ConfigType]interface{}{
 				artifacts.WarConfigType: artifacts.WarArtifactConfig{
-					DeploymentFile: filepath.Base(path),
-					JavaVersion:    t.WarConfig.JavaVersion,
+					DeploymentFilePath: relPath,
```

Comment:
Why do we need path relative to source directory in this config? the paths in the artifact already have the right paths to place the Dockerfile. In here the paths are relative to the paths in the Paths section of the artifact, in this case, its just the archive name

Reply:
see above

Reply:
We seem to be computing the path relative to input directory and not service directory.